compiler/catala_utils/message.ml

```diff
@@ -459,6 +458,26 @@ let join_pos ~pos ~fmt_pos ~extra_pos =
     Some pos
   | _ -> None
 
+let warning
```

Comment:
Here you add a warning function but you do not use it in the rest of the compiler codebase in this PR? There are already some warnings thrown by the compiler (for instance in the linting module), I would have expected this PR to reactor them so that they're taken into account correctly by the LSP.

Reply:
We already have an existing generic processing for messages to convert them into a reasonable format for LSPs. We can tweak it if needed but from what I saw it seemed to output a decent message:
![image](https://github.com/user-attachments/assets/e5f318e5-ed5c-4afe-9f0e-331c3621ce09)

Reply:
To make it clearer, every existing warning in the code base (e.g. the linter) will go through this function which will notify the message handler (for which we defined a hook in the LSP) that returns a printable content that was preprocessed to display a "vscode-friendly" message (because we also set the formatting mode to LSP).